client-js/app/elements/labrad-registry.ts

```diff
@@ -155,21 +192,29 @@ export class LabradRegistry extends polymer.Base {
       return;
     }
 
+    // New keys have a textbox, so we only want to submit on Shift+Enter,
+    // not on a solo enter keypress.
+    if (event.detail.combo !== 'shift+enter') {
+      if (dialog.id == 'newKeyDialog' || dialog.id === 'editValueDialog') {
```

Comment:
I would do the dialog check first, since the comment notes that this is a special case for those dialogs. Also, the comment should refer to both affected dialogs, or just be generic: `// For dialogs with a textbox, ...`